cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodegroupset
+
+import (
+	"math"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+)
+
+const (
+	// MaxAllocatableDifferencePercent describes how Node.Status.Allocatable can differ between
+	// groups in the same NodeGroupSet
+	MaxAllocatableDifferencePercent = 0.05
+	// MaxFreeDifferencePercent describes how free resources (allocatable - daemon and system pods)
+	// can differ between groups in the same NodeGroupSet
+	MaxFreeDifferencePercent = 0.05
+)
+
+func compareResourceMapsWithTolerance(resources map[apiv1.ResourceName][]resource.Quantity,
```

Comment:
It's a private helper function, so I just passed the data structure I already had. As to why not use 2 maps in the first place - I don't think it matters much for temporary datastructure, that is never passed outside and this was just more convenient to create with a loop. I can refactor it if you think it's a bad choice.